### PR TITLE
kraken: make data copyable and add method on data_manager for handling disruptions

### PR DESCRIPTION
**Pull request by kinnou02**
for now we do a copy of data every time when we need it

### CODE REVIEW

File: source/type/data.cpp [hunk 1]
```diff
@@ -68,6 +68,19 @@ Data::Data() :
     this->loaded = false;
 }
 
+Data::Data(const Data& other):
+    meta(std::make_unique<MetaData>(*other.meta)),
+    pt_data(std::make_unique<PT_Data>(*other.pt_data)),
+    geo_ref(std::make_unique<navitia::georef::GeoRef>(*other.geo_ref)),
+    dataRaptor(std::make_unique<navitia::routing::dataRAPTOR>(*other.dataRaptor)),
+    fare(std::make_unique<navitia::fare::Fare>(*other.fare)),
+    find_admins([&](const GeographicalCoord &c){return geo_ref->find_admins(c);})
```
Comment (antoine-de) [inline source/type/data.cpp]: why don't you copy the function too ?

File: source/type/pt_data.h [hunk 2]
```diff
@@ -44,7 +44,7 @@ namespace navitia { namespace type {
 // En attendant on utilise std::map car on sait le sérialiser...
 typedef std::map<std::string, idx_t> ExtCodeMap;
 
-struct PT_Data : boost::noncopyable{
+struct PT_Data{
```
Comment (TeXitoi) [inline source/type/pt_data.h:47]: No object in PT_Data is copied!!! the two copies point to the same objects! And they will be freed!

File: source/type/data.h [hunk 3]
```diff
@@ -126,6 +126,7 @@ class Data : boost::noncopyable{
 
     Data();
     ~Data();
+    Data(const Data& other);
```
Comment (TeXitoi) [inline source/type/data.h:129]: implicit copy assignment will be generated and will be wrong. delete it or write it.

File: source/type/data.cpp [hunk 4]
```diff
@@ -68,6 +68,19 @@ Data::Data() :
     this->loaded = false;
 }
 
+Data::Data(const Data& other):
+    meta(std::make_unique<MetaData>(*other.meta)),
+    pt_data(std::make_unique<PT_Data>(*other.pt_data)),
+    geo_ref(std::make_unique<navitia::georef::GeoRef>(*other.geo_ref)),
```
Comment (TeXitoi) [inline source/type/data.cpp:74]: object managed by GeoRef (and maybe others) will be freed!

**Comment by TeXitoi** [comment]
If the copy only takes a minute, it is because almost nothing is copied. We need to remove every new and delete in the code to be sure to be correct. (using std::unique_ptr everywhere)

**Pull request merged**